### DISCUSSION TITLE
Renovate automatically processing repos after upgrade to Renovate CE (Docker image 6.0.0)

### DISCUSSION THREAD

**Discussion by enricojonas** [mend/renovate-ce-ee#340] [Q&A] [answered]
After migrating to Renovate CE (Docker image 6.0.0-full) it seems that after successful startup renovate does a processing of all repositories for the entire org (running on GHES).

Following environment variables are set in our setup:

```
MEND_RNV_ACCEPT_TOS = "y"
MEND_RNV_PLATFORM = github (value did not change)
MEND_RNV_ENDPOINT = (value did not change)
MEND_RNV_GITHUB_APP_ID = (value did not change)
MEND_RNV_GITHUB_APP_KEY = (value did not change)
GITHUB_COM_TOKEN = (value did not change)
MEND_RNV_LICENSE_KEY = (value did not change)
MEND_RNV_WEBHOOK_SECRET = webhook secret as configured in GHES APP (value did not change)
MEND_RNV_CRON_JOB_SCHEDULER = "0 5 * * *"
LOG_LEVEL = DEBUG
LOG_FORMAT = "json"
```

Following settings in config.js (no change was done)

```
    "autodiscover": false,
    "requireConfig": "optional",
    "onboarding": false,
    "branchPrefix": "renovate-",
    "dependencyDashboard": true,
    "dependencyDashboardAutoclose": true,
    "dependencyDashboardFooter": "\n- [ ] <!-- manual job -->Check this box to trigger a request for Renovate to run again on this repository\n",
    "prCreation": "approval",
    "branchConcurrentLimit": 0,
    "vulnerabilityAlerts": {
        "prConcurrentLimit": 1
    },
```

I can see the following in the logs:

```
"Server listening on port 8080"
"Running Renovate job"
"Executing Renovate run command"
--> reason: "hourly"

"Starting scheduler" - cronTime: "0 5 * * *"
```

I verified that this does not happen with v5.0.0 - it was just running according to schedule or on incoming webhook. Last message in log used to be that the server is listening.

Processing all repositories takes a long time.

**Comment by rarkins** [comment]
I believe it's a one-time thing to run a job for each repo once the first time it's detected.

@Gabriel-Ladzaretti is a job run for each repo on every startup, or only when it's _added to the DB_?

**Comment by Gabriel-Ladzaretti** [comment]
It is the former.
At startup we do a sync and then enqueue a job for all repos.

**Comment by enricojonas** [comment]
Ok so this is a change compared to the old setup.

It would be nice if there would be a config to skip those initial jobs as it keeps renovate busy for 15 minutes (for a rather small org) and does not add much value when you run renovate in K8S. The pod can be re-created any time - moved to other node etc. It would put unnecessary load on GHES and other systems.

Any thoughts?

**Reply by rarkins** [reply]
@Gabriel-Ladzaretti can we make a setting like "do jobs on startup" defaulting to false?

**Reply by Gabriel-Ladzaretti** [reply]
Sure. Just note that sync will result in a job for every new repo discovered regardless.

**Reply by enricojonas** [reply]
That would be good.

What about "autodiscover": false setting that we have - shouldn't this prevent this?

**Comment by enricojonas** [comment, marked as answer]
Hi,

thanks for implementing this, MEND_RNV_ENQUEUE_JOBS_ON_STARTUP = "disabled" is working for us now :+1:
